src/functions.rs

```diff
@@ -477,6 +583,24 @@ pub mod stdlib {
     pub mod list {
         use super::*;
 
+        /// The implementation of the standard [`concat`][`crate::reference::functions#concat`] function.
+        pub struct Concat;
+
+        impl Function for Concat {
+            fn call(
+                &self,
+                _graph: &mut Graph,
+                _source: &str,
+                parameters: &mut dyn Parameters,
+            ) -> Result<Value, ExecutionError> {
+                let mut result = Vec::new();
+                while let Ok(list) = parameters.param() {
+                    result.append(&mut list.into_list()?);
```

Comment:
Very nice, I've always done this with `extend_from_slice`, didn't realize there was an `append` method!